pkg/asset/cluster/aws/aws.go

```diff
@@ -55,27 +55,45 @@ func tagSharedVPCResources(ctx context.Context, clusterID string, installConfig
 		return err
 	}
 
-	ids := make([]*string, 0, len(privateSubnets)+len(publicSubnets))
-	for id := range privateSubnets {
-		ids = append(ids, aws.String(id))
-	}
-	for id := range publicSubnets {
-		ids = append(ids, aws.String(id))
-	}
-
 	session, err := installConfig.AWS.Session(ctx)
 	if err != nil {
 		return errors.Wrap(err, "could not create AWS session")
 	}
+	ec2Client := ec2.New(session, aws.NewConfig().WithRegion(installConfig.Config.Platform.AWS.Region))
+
+	ids := make([]*string, 0, len(publicSubnets))
+	for id := range publicSubnets {
+		ids = append(ids, aws.String(id))
+	}
 
+	// Add tags for Public Subnets
 	tagKey, tagValue := sharedTag(clusterID)
+	publicElbTagKey, publicElbTagValue := publicElbTag()
```

Comment:
As this is the only place publicElbTag() is used, perhaps these variables could be simplified by statically assigning the values here.

Reply:
I just tried that. Key is expected to be a *string and a static string did not get converted. So, will have to keep this indirection for now.